### DISCUSSION TITLE
Questions about some of the api methods

### DISCUSSION THREAD

**Discussion by gabrielcarloto** [Flow-Launcher/Flow.Launcher#1305] [Q&A] [answered]
I was looking at the [JSON RPC Introduction](https://www.flowlauncher.com/docs/#/json-rpc) and the [api](https://github.com/Flow-Launcher/Flow.Launcher/blob/master/Flow.Launcher.Plugin/Interfaces/IPublicAPI.cs) and decided to try the methods. Here are my questions:

### CopyToClipboard

This is actually the only one that worked the way I thought it would. The question is: why isn't this method listed in the docs?

### ChangeQuery

Initially, I thought it would work like the Explorer plugin: I type the query, press enter and it changes the query without closing the query box. But what happens is that it closes the query box and, if I set the `requery` to true, when I launch Flow again the query I sent from the method ChangeQuery is there. Is there a way I can make it behave like the Explorer plugin.

### ShellRun, GetTranslation, OpenSettingDialog and Start / StopLoadingBar

Nothing happened at all. Maybe I did something wrong, specially when testing the GetTranslation. Also, I couldn't find Start / StopLoadingBar in the [api](https://github.com/Flow-Launcher/Flow.Launcher/blob/master/Flow.Launcher.Plugin/Interfaces/IPublicAPI.cs).

**Comment by Garulf** [comment, marked as answer]
> ### CopyToClipboard
> 
> 
> 
> This is actually the only one that worked the way I thought it would. The question is: why isn't this method listed in the docs?
> 
> 
This a new API added in the last version. Unfortunately it has yet to be added to the docs. 


> ### ChangeQuery
> 
> 
> 
> Initially, I thought it would work like the Explorer plugin: I type the query, press enter and it changes the query without closing the query box. But what happens is that it closes the query box and, if I set the `requery` to true, when I launch Flow again the query I sent from the method ChangeQuery is there. Is there a way I can make it behave like the Explorer plugin.
> 
You'll want to set 'dontHideAfterAction' to 'true' under 'JsonRPCAction'

> ### ShellRun, GetTranslation, OpenSettingDialog and Start / StopLoadingBar
> 
> 
> 
> Nothing happened at all. Maybe I did something wrong, specially when testing the GetTranslation. Also, I couldn't find Start / StopLoadingBar in the [api](https://github.com/Flow-Launcher/Flow.Launcher/blob/master/Flow.Launcher.Plugin/Interfaces/IPublicAPI.cs).

Not sure about 'GetTranslation' but the rest *should* work. IIRC the loading bar ones work under methods that have been triggered to run when the user selects an item.

**Reply by taooceros** [reply]
`GetTranslation` shall not work for jsonrpc plugin since there's currently no way to communicate back to jsonrpc plugin.

**Reply by taooceros** [reply]
I suspect whether Start/StopLoadingBar may work because the visibility of the progressbar is triggered at every query starting or end, so maybe that's not applicable for jsonrpc plugins since everytime they triggers an api will require a query response.

**Reply by taooceros** [reply]
Take a look on #937 if we agree to expand the communication model to bidirectional.
